File: leetcode/392.is-subsequence.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isSubsequence = function(s, t) {
  let p = 0
  for (const c of t) {
    if (c === s[p]) {
      p++
    }
    if (p === s.length) {
      return true
    }
  }
  return p === s.length // fix s === ''  t === ''
}

File: leetcode/392.is-subsequence.go
package leetcode
func isSubsequence(s string, t string) bool {
	if len(s) == 0 {
		return true
	}
	var pos int
	for i := 0; i < len(t); i++ {
		if s[pos] == t[i] {
			pos++
		}
		if pos == len(s) {
			return true
		}
	}
	return false
}
